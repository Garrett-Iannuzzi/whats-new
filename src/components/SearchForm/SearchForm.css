aside {
  background-color: darkgrey;
  border-bottom: solid 4px white;
  display: flex;
  height: 3.2em;
  justify-content: space-between;
}

.news-type-btn {
  height: 2em;
  font-weight: bold;
  margin: 20px 10px;
  width: 15%;
}

/* .navbar {
  background-color: darkgrey;
  border-bottom: solid 4px white;
  display: flex;
  height: 3.2em;
  justify-content: space-between;
} */

h1 {
  font-size: 1.7em;
  color: black;
  padding: 15px 0px;
}