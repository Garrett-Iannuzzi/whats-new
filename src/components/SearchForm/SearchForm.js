import React from 'react';
import './SearchForm.css';

const SearchForm = () => {
  return (
    <aside>
      <button className='news-type-btn'>Local News</button>
      <button className='news-type-btn'>Tech</button>
      <button className='news-type-btn'>Entertainment</button>
      <button className='news-type-btn'>Science</button>
      <button className='news-type-btn'>Health</button>
    </aside>
  )
}
export default SearchForm;