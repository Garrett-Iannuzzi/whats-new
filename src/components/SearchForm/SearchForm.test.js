import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('Search Form', () => {

  let event = { keyCode: 13, target: { name: 'input', value: 'test value' } };
  let wrapper
  let handleSearchMock

  beforeEach(() => {
    handleSearchMock = jest.fn();
    wrapper = shallow(<SearchForm 
      handleSearch={handleSearchMock}
    />);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the state on the input', () => {
    wrapper.instance().handleChange(event)
    expect(wrapper.state('inputValue')).toEqual('test value')
  });

  it('should call handleSearch when the search button is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(handleSearchMock).toHaveBeenCalled()
  });

  it('should run handleEnterKey when the enter key is pressed', () => {
    wrapper.instance().handleEnterKey(event)
    expect(handleSearchMock).toHaveBeenCalled()
  });
});