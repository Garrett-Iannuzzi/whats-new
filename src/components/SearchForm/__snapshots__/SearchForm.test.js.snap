// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form should match the snapshot with all data passed in correctly 1`] = `
<div
  className="search-bar-div"
>
  <input
    aria-label="Text Box"
    onChange={[Function]}
    placeholder="Search News"
    type="text"
    value=""
  />
  <button
    className="search-btn"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
